lib/node_modules/@stdlib/_tools/scaffold/math-special-unary/scripts/generate_files.js

```diff
@@ -0,0 +1,455 @@
+			} else {
+				cOut += '\n';
+			}
+		}
+		cOut += '\n';
+	}
+	cOut += '};\n\n';
+	cOut += '// Define a list of ndarray function "data" (in this case, callbacks):\n';
+	cOut += 'static void *data[] = {\n';
+	functionIndex = 0;
+	for ( i = 0; i < result.inputTypes.length; i++ ) {
+		inputType = result.inputTypes[ i ];
+		outputTypes = grouped[ inputType ];
+
+		cOut += '\t// ' + inputType + ' (' + outputTypes.length + ')\n';
+		for ( j = 0; j < outputTypes.length; j++ ) {
+			cOut += '\t(void *)' + matches[ functionIndex ][ 6 ];
+			functionIndex += 1;
+			if ( functionIndex < matches.length ) {
+				cOut += ',\n';
+			} else {
+				cOut += '\n';
+			}
+		}
+		cOut += '\n';
+	}
+	cOut += '};\n\n';
+	cOut += '// Create an ndarray function object:\n';
+	cOut += 'static const struct ndarrayFunctionObject obj = {\n';
+	cOut += '\t// ndarray function name:\n';
+	cOut += '\tname,\n\n';
+	cOut += '\t// Number of input ndarrays:\n';
+	cOut += '\t1,\n\n';
+	cOut += '\t// Number of output ndarrays:\n';
+	cOut += '\t1,\n\n';
+	cOut += '\t// Total number of ndarray arguments (nin + nout):\n';
+	cOut += '\t2,\n\n';
+	cOut += '\t// Array containing ndarray functions:\n';
+	cOut += '\tfunctions,\n\n';
+	cOut += '\t// Number of ndarray functions:\n';
+	cOut += '\t' + matches.length + ',\n\n';
+	cOut += '\t// Array of type "numbers" (as enumerated elsewhere), where the total number of types equals `narrays * nfunctions` and where each set of `narrays` consecutive types (non-overlapping) corresponds to the set of ndarray argument types for a corresponding ndarray function:\n';
+	cOut += '\ttypes,\n\n';
+	cOut += '\t// Array of void pointers corresponding to the "data" (e.g., callbacks) which should be passed to a respective ndarray function (note: the number of pointers should match the number of ndarray functions):\n';
+	cOut += '\tdata\n';
+	cOut += '};\n\n';
+	cOut += 'STDLIB_NDARRAY_NAPI_MODULE_UNARY( obj )\n';
+	return cOut;
+}
+
+
+// MAIN //
+
+/**
+* Generates scaffold files for a math special unary function.
+*
+* @param {string} basePkg - base package name
+* @param {string} destDir - destination directory
+*/
+function generateFiles( basePkg, destDir ) {
+	var filteredMatches;
+	var typesJsonOut;
+	var dataOut;
+	var matches;
+	var header;
+	var jsOut;
+	var cOut;
+	var i;
+
+	// Generate and filter matches table:
+	matches = generateDtypes( basePkg );
+
+	// Filter out generic types for addon.c:
+	filteredMatches = [];
+	for ( i = 0; i < matches.length; i++ ) {
+		if ( matches[ i ][ 0 ] !== 'generic' && matches[ i ][ 1 ] !== 'generic' ) {
+			filteredMatches.push( matches[ i ] );
+		}
+	}
+
+	// Generate license header:
+	header = licenseHeader( 'Apache-2.0', 'js', {
+		'year': currentYear(),
+		'copyright': 'The Stdlib Authors'
+	});
+	header += '\n/* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */\n';
+
+	// Generate types.js:
+	jsOut = generateTypesFile( matches, header );
+	writeFileSync( join( destDir, 'lib/types.js' ), jsOut, {
```

Comment:
You make heavy use of string concatenation to generate entire files, including license headers. I wonder if we could use template files with handlebar template strings `{{}}` similar to what we do for other files. That might reduce the visual noise in this file.